### ISSUE TITLE
ol.control.ScaleLine regression with EPSG:4326

### ISSUE THREAD

**Issue by pedros007** [openlayers/openlayers#7694]
I have a map with `projection: 'EPSG:4326'`.  In v4.5.0, `ol.control.ScaleLine` seems to work. [this codepen with v4.5.0](https://codepen.io/pschmitt/pen/VyxWjN) shows the initial value of the scale line = `5,000 km`.  At maximal zoom, I see the scale line reports 50mm ~(although this seems to be reported at max zoom regardless of latitude)~.

[A fork of that codepen using v4.6.0](https://codepen.io/anon/pen/opdWNe) shows the initial value of the scale line = `500,000,000km`.  At maximal zoom, I see the scale line reports 5km (regardless of latitude).  I also see the same behavior when using v4.6.4.

v4.6.4 seems to work as expected if I comment out these lines from #7462:

```javascript
 if (units != ol.control.ScaleLineUnits.DEGREES) {
    pointResolution *= projection.getMetersPerUnit();
  }
```

**Comment by pedros007** [comment]
A little history I found:
1. #7087 added variable scale depending on latitude for EPSG:4326
2. It seems #7087 introduced a regression for some users.  See #7309, where scaleLine values were wrong for certain projections (such as [EPSG:27700](https://epsg.io/27700), which is reported in meters)
3. #7462 attempted to fix this, but introduced a regression for maps having `projection: 'EPSG:4326'` (aka this issue)

**Comment by pedros007** [comment]
Should the conditional on [src/ol/control/ScaleLine.js#180](https://github.com/openlayers/openlayers/blob/ece1d4d231cbb6e78805320596dbe1b76090f05c/src/ol/control/ScaleLine.js#L179-L181) be more complex, something like the following?

```javascript
  if (projection.units == ol.proj.Units.METERS && units == ol.control.ScaleLineUnits.METRIC) {
    pointResolution *= projection.getMetersPerUnit();
  }
```

**Comment by ahocevar** [comment]
Such a change would make sense to me. Can you create a pull request please?

**Comment by Docteur-RS** [comment]
For people wanting to fix this in v4.6.5 you can follow [this PR](https://github.com/openlayers/openlayers/pull/7700/files/500a27bd8bf76d49dd82410dd1e4df6477cae460). Unfortunatly its for v5.* so you can just copy paste it.

The exact fix is :
`if (projection.getUnits() != _ol_control_ScaleLineUnits_.DEGREES && units == 
 ol_control_ScaleLineUnits_.METRIC)
pointResolution *= projection.getMetersPerUnit();`

This should replace the condition used on [line 180] in(https://github.com/openlayers/openlayers/blob/ece1d4d231cbb6e78805320596dbe1b76090f05c/src/ol/control/ScaleLine.js#L179-L181) scaleline.js
